mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/impl/PojoMassIndexingBatchCoordinator.java

```diff
@@ -96,12 +99,25 @@ public void runWithFailureHandler() throws InterruptedException {
 	 * Operations to do before the multiple-threads start indexing
 	 */
 	private void beforeBatch() throws InterruptedException {
-		// Create an agent to suspend concurrent indexing
-		agent = mappingContext.createMassIndexerAgent(
-				new PojoMassIndexerAgentCreateContextImpl( mappingContext, detachedSession.tenantIdentifier() )
-		);
+		// Prepare the contexts first. These will be used for all batch related work:
+		for ( DetachedBackendSessionContext detachedSession : detachedSessions ) {
+			coordinatorContexts.add(
+					new SessionContext(
+							// Create an agent to suspend concurrent indexing
+							mappingContext.createMassIndexerAgent(
+									new PojoMassIndexerAgentCreateContextImpl( mappingContext, detachedSession.tenantIdentifier() )
+							),
+							pojoScopeDelegate.workspace( detachedSession ),
+							detachedSession
+					)
+			);
+		}
+
 		// Start the agent and wait until concurrent indexing actually gets suspended
-		Futures.unwrappedExceptionGet( agent.start() );
+		applyToAllContexts(
+				c -> CompletableFuture.supplyAsync( () -> c.agent().start() )
```

Comment:
so this line was a problem 🥲 it should've just being:
```suggestion
				c -> c.agent().start()
```
as `agent.start()` already returns a future

Reply:
Right, fixed